src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,144 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private T[] elements;
+    private int size = 0;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void ensureCapacity() {
+        T[] newElements = (T[]) new Object[elements.length * 2];
+        copyArray(elements, 0, newElements, 0, size);
+        elements = newElements;
+    }
+
+    private void copyArray(T[] arrayFrom,
+                           int arrayFromPos,
+                           T[] arrayTo,
+                           int arrayToPos,
+                           int amount) {
+        for (int i = 0; i < amount; i++) {
+            arrayTo[arrayToPos + i] = arrayFrom[arrayFromPos + i];
```

Comment:
Use System.arraycopy() to move your array elements. This is a more efficient and recommended way to copy arrays in Java.